src/utils/word-api.ts

```diff
@@ -17,42 +24,39 @@ class WordApi {
     });
   }
 
-  static async getJabRefCitations(): Promise<Array<Word.ContentControl> | void> {
-    return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
-      await context.sync();
-      return contentControls.items.filter(
-        (citation) =>
-          citation.tag.includes("JABREF-CITATION") &&
-          !citation.tag.includes("NEW")
-      );
-    }).catch((error) => {
-      console.log(`Error: ${JSON.stringify(error)}`);
-      if (error instanceof OfficeExtension.Error) {
-        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
-      }
-    });
+  static async getJabRefCitations(
+    context: Word.RequestContext
+  ): Promise<Array<Word.ContentControl> | void> {
+    const { contentControls } = context.document;
+    context.load(contentControls, "tag, length");
+    await context.sync();
+    return contentControls.items.filter((citation) =>
+      citation.tag.includes("JABREF-CITATION")
+    );
   }
 
   static async getPositionOfNewCitation(): Promise<number | void> {
     return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
+      const { contentControls } = context.document.body;
+      context.load(contentControls, "length, items");
+      const currentposition = context.document.getSelection();
       await context.sync();
-      let length = 0;
-      let pos = 0;
-      for (let i = 0, ilen = contentControls.items.length; i < ilen; i += 1) {
-        const { tag } = contentControls.items[i];
-        if (tag.includes("JABREF-CITATION")) {
-          if (tag.substring(16) === "NEW") {
-            pos = length;
-            break;
-          }
-          length += 1;
+      const jabRefCitations = contentControls.items.filter((citation) =>
+        citation.tag.includes("JABREF-CITATION")
+      );
+      const locationArray = jabRefCitations.map((citation) => {
+        const citationToCompareWith = citation.getRange("Start");
+        const currentSelectionRange = currentposition.getRange("Whole");
+        return citationToCompareWith.compareLocationWith(currentSelectionRange);
+      });
+      await context.sync();
+      for (let i = 0; i < locationArray.length; i += 1) {
```

Comment:
Why not use find to search for the index with name "After" ? 

const position = locationArry.find(idx=> idx === "After")
...



https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Which could also be concatenated with above call, i.e. directly after `map(...)`